### ISSUE TITLE
installation example gist & LongTermAccessToken question

### ISSUE THREAD

**Issue by rwitchell** [knpuniversity/oauth2-client-bundle#26]
Hi Ryan,

So I've been meaning to showcase my installation as a gist for other users to help people understand how to install this.
It's not easy to install, and there are so many different variances that come into play when installing.

I've also posted it for some help - i'm trying to understand where in my code i would request my longTermAccessToken?
Currently: the user can log in; they get the my registration page so i can get additional information; and i'm saving all of this into the database. The only value i'm not saving is the LongTermAccessToken in the database.
I'm also wondering how often i should re-request the LongTermAccessToken / and on what screen?

I'll be happy to move this gist into this repo's main readme.md as an additional installation example.

https://gist.github.com/rwitchell/f380be61d6d738c78ab567e79b04ffa7

**Comment by weaverryan** [comment]
Hey @rwitchell!

Well, I'm happy you posted this. I agree about there being many variances for install, and I'd definitely like your help making this easier. Using this bundle *should* be easy. After all, you're really in almost complete control of your code. That being said, yes, I know it's not always that easy in reality :).

First, about the "long lived access token". Facebook handles this a little bit differently than some other services (first, it's a "long-lived token" instead of the normal "refresh token"). I believe that you should request the long lived token immediately after you retrieve the access token (so, in the authenticator when using Guard). And I would save this to the database.

When should you "refresh" this long-lived token? Well, that's tougher, and it's not something I've head to deal with. But, refreshing the long-lived token requires sending them back through the OAuth flow (to get another short-lived token). You could auto-refresh it by exchanging the short-lived token for a long-lived one every time they log into your site via Facebook. But eventually, if that token expires, and they are not actively on your site, I don't think there's anything you can do. Additionally, when they login via the normal form login, if you detect they connected with Facebook and you'd like to refresh their long-lived token, you *could* redirect them immediately to Facebook right then. In theory, they wouldn't even notice this... unless they're not logged into Facebook.

So basically, I'm not too sure about the long-lived token stuff. It's tough because those tokens *will* expire you *must* send the client back through the OAuth process to refresh it. Fortunately, this is an issue shared by anyone using Facebook (not just people of this bundle, or even PHP people), so you will hopefully be able to find some more help out there :).

About your setup, are you using FOSUserBundle? And are you really using the `form_login` functionality (I noticed it in your firewall, but I also noticed your own custom form login authenticator). What parts were most confusing? What were your biggest "I don't know how to do X" or "I don't know if I'm doing Y correctly" moments? We can use that to "beef up" the docs :).

Cheers!
